twine/commands/upload.py

```diff
@@ -48,6 +50,40 @@ def skip_upload(response, skip_existing, package):
             (response.status_code == 403 and msg_403 in response.text)))
 
 
+def check_status_code(response, verbose):
```

Comment:
Why did we move this out of `utils`? I believe it was in there as part of a foundation for a public API

Reply:
You've presented a dichotomy without explaining your reasoning @bhrutledge. Why are those the only two options in your mind?

Reply:
Hm. Those were the two that occurred to me, without thinking about ones that might involve more substantial changes. Happy to consider others.

Reply:
I'm looking for _why_ you think these functions should move in the first place in either direction.

Reply:
Ah. They seem related, in that they both check the response from `upload`.

I also see that I could have been clearer in my original comment. Let me try again:

Is it okay to leave it here? Or should we move it back? If the latter, should we consider moving `skip_upload` to `utils`?

Reply:
It's fine to leave it here for now, that said, if and when Twine grows more responsibilities, I imagine that this will be moved back to utils because it will become more generally useful. If and when that happens, that won't make `skip_upload` belong in utils because it's highly specific to uploading files.

Right now, locality is important although we're moving something purely for that purpose instead of it having been organized correctly from the start. That adds confusion and makes things harder to review in general when conflating moving something in a commit or PR with other work. The bar needs to be higher in a project like this for moving things around. 

You still seem to think "Either `check_status_code` belongs here or both that and `skip_upload` belong in `utils`." and I'd like to emphatically encourage you to stop thinking in that binary mode as a new maintainer of this project. Further, there are projects that (against our advice) import liberally from the twine namespace. That means someone could also be importing this function from utils so the question you should be answer is "What's the least surprising thing to do?"

Reply:
Thanks for detailed response, @sigmavirus24. I see how my initial suggestion of moving `check_status_code` was hasty, and I agree that it would have been better to do that in separate PR, if at all. I'll be more conservative with my suggestions in the future, and mention the other maintainers earlier.

That said, in general, I try to present my ideas as possibilities, and indicate openness to alternatives. So, in this case, I don't feel strongly that `check_status_code` and `skip_upload` should be siblings.

FWIW, it were up to me, I would move `check_status_code` back to `utils`, for the reasons you outlined in your comment.

Reply:
Hi @bhrutledge @sigmavirus24, just as a final clarification, do you folks think it's best for me to leave it here for now?

Reply:
If you're game to move it back, I think that's a good idea. My apologies for sending you down the wrong path.

Reply:
No worries! I added two tests - one that calls `upload` and another that calls `check_status_code`. Should I split them up and keep one in `test_upload` and the other in `test_utils`? They're both in `test_upload currently`.

Reply:
Ah, good catch. I think it'd be ideal to move the one for `check_status_code` to `test_utils`.

Also, if you move `check_status_code` back to `utils`, it'd be ideal to put it in the same spot that it was originally, to minimize the size of the diff.